linode/manager
# change, test: [M3-7073, M3-7176] - Remove `dcSpecificPricing` and `objDcSpecificPricing` feature flags

## Description 📝
Both phases of feature have been released. We want to make our changes permanent by removing the feature flag logic from our code.

## Changes  🔄
List any change relevant to the reviewer.
- Removes the `dcSpecificPricing` and `objDcSpecificPricing` feature flags from `/src` code, updating component logic and utils as necessary
- Updates a handful of unit tests to remove feature flag logic
- Updates the following e2e tests to remove feature flag logic:
   - `volumes/create-volume.spec.ts`
   -  `linodes/migrate-linode.spec.ts`
   - `linodes/create-linode.spec.ts`
   - `linodes/backup-linode.spec.ts`
   - `linodes/clone-linode.spec.ts`
   - `kubernetes/lke-update.spec.ts`
   - `kubernetes/lke-create.spec.ts`
   - `objectStorage/enable-object-storage.spec.ts`
   - `nodebalancers/smoke-create-nodebal.spec.ts`
   -  `billing/billing-invoices.spec.ts`

## Preview 📷

| Before  | After   |
| ------- | ------- |
|![Screenshot 2023-11-09 at 8 41 34 AM](https://github.com/linode/manager/assets/114685994/7be850aa-d8f9-455c-a6c8-cf377fc3f513) |![Screenshot 2023-11-09 at 8 18 31 AM](https://github.com/linode/manager/assets/114685994/3ea81b04-a4b0-4dd9-ae09-d0b7f7534bf2) |

## How to test 🧪

### Prerequisites
(How to setup test environment)
- Check out this PR.
- As part of testing these flows, you'll want to create resources in `Jakarta, ID` or `Sao Paulo, BR` and compare that to a region without DC-specific prices (e.g. `Newark, NJ`).

### Verification steps 
(How to verify changes)
- Verify that the feature flags are not listed in the dev tools and are now returning `false` via Launch Darkly in the development environment (check the network requests)
- Verify that the following features are all still visible:
   - In the Kube Create flow and Linode create/clone flows region selection: DC Specific Pricing Notices and Docs Links ([#9572](https://github.com/linode/manager/pull/9572))
   - For invoices from **October** onwards, a region column displays. Otherwise, the region column is hidden. ([#9597](https://github.com/linode/manager/pull/9597))
   - In the Linode Create flows, prices are not displayed until a region selection is made ([#9598](https://github.com/linode/manager/pull/9598))
   - In the Linode migration flow, a price display is shown if there is a difference in migration-from and migration-to region pricing ([#9570](https://github.com/linode/manager/pull/9570))
   - In the Kube Create flow, prices are not displayed until a region selection is made and on the Kube details page, dynamically calculated prices are displaying correctly ([#9606](https://github.com/linode/manager/pull/9606))
   - With a linode on your account in Jakarta or Sao Paulo, Monthly Network Transfer Pool dialog and linode usage displays include a DC-specific transfer pool ([#9620](https://github.com/linode/manager/pull/9620)), ([#9692](https://github.com/linode/manager/pull/9692))
   - In the NodeBalancer create flow, prices are calculated dynamically and displaying correctly ([#9566](https://github.com/linode/manager/pull/9566))
   - Add DC-specific pricing to Kubernetes HA ([#9568](https://github.com/linode/manager/pull/9568))
   - For Linode Backups (Create flow, Linode Details page, Enable Backups drawer), prices are calculated dynamically and displaying correctly ([#9588](https://github.com/linode/manager/pull/9588))
   - In the Volumes Create flows (from Volumes Landing and Linode Details), prices are calculated dynamically and displaying correctly ([#9569](https://github.com/linode/manager/pull/9569)), ([#9683](https://github.com/linode/manager/pull/9683)) 
   - In the OBJ Create flows (access key and buckets), the Create Bucket drawer displays correct overages and the Enable Object Storage modal contains correct, non-beta copy ([#9813](https://github.com/linode/manager/pull/9683))
- Verify that all unit tests pass
- Verify that all e2es listed above pass

Confirm the following e2es pass:
```
yarn cy:run -s "cypress/e2e/core/volumes/create-volume.spec.ts,cypress/e2e/core/linodes/migrate-linode.spec.ts,cypress/e2e/core/linodes/create-linode.spec.ts,cypress/e2e/core/linodes/backup-linode.spec.ts,cypress/e2e/core/linodes/clone-linode.spec.ts,cypress/e2e/core/kubernetes/lke-update.spec.ts,cypress/e2e/core/kubernetes/lke-create.spec.ts,cypress/e2e/core/objectStorage/enable-object-storage.spec.ts,cypress/e2e/core/nodebalancers/smoke-create-nodebal.spec.ts,cypress/e2e/core/billing/billing-invoices.spec.ts"
```

## As an Author I have considered 🤔

*Check all that apply*
